src/parse/import.rs

```diff
@@ -134,7 +104,67 @@ impl<'a> Parser<'a> {
             }
             .parse();
         }
+        self.whitespace();
 
         Err(("Can't find stylesheet to import.", span).into())
     }
+    pub(super) fn import(&mut self) -> SassResult<Vec<Stmt>> {
+        self.whitespace();
+        match self.toks.peek() {
+            Some(Token { kind: '\'', .. })
+            | Some(Token { kind: '"', .. })
+            | Some(Token { kind: 'u', .. }) => {}
+            Some(Token { pos, .. }) => return Err(("Expected string.", *pos).into()),
+            None => return Err(("expected more input.", self.span_before).into()),
+        };
+        let Spanned {
+            node: file_name_as_value,
+            span,
+        } = self.parse_value(true)?;
+
+        match file_name_as_value {
+            Value::String(s, QuoteKind::Quoted) => {
+                if s.ends_with(".css") || s.starts_with("http://") || s.starts_with("https://") {
+                    Ok(vec![Stmt::Import(format!("\"{}\"", s))])
+                } else {
+                    self._parse_single_import(&s, span)
+                }
+            }
+            Value::String(s, QuoteKind::None) => {
+                if s.starts_with("url(") {
+                    Ok(vec![Stmt::Import(s)])
+                } else {
+                    self._parse_single_import(&s, span)
+                }
+            }
+            Value::List(v, Comma, _) => {
+                let mut list_of_imports: Vec<Stmt> = Vec::new();
+                for file_name_element in v {
+                    match file_name_element {
+                        Value::String(s, QuoteKind::Quoted) => {
+                            if s.ends_with(".css")
+                                || s.starts_with("http://")
+                                || s.starts_with("https://")
+                            {
+                                list_of_imports.push(Stmt::Import(format!("\"{}\"", s)));
+                            } else {
+                                list_of_imports.append(&mut self._parse_single_import(&s, span)?);
+                            }
```

Comment:
This part seemed duplicated, could split it into a function.

Reply:
Yes the code for pattern matching/parsing a single import is duplicated but it is only done duplicated once. And as the code won't be needed to be reused anywhere else I am going to keep it simple and not split it up. However can do if anyone wants it split up :)